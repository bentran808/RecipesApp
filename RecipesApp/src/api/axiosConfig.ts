import axios from 'axios';

export const axiosInstance = axios.create({
  baseURL: 'http://10.0.3.2:3000', //Change to http://10.0.2.2 for AVD and http://localhost for iOS
  headers: {
    'Content-Type': 'application/json'
  }
});
