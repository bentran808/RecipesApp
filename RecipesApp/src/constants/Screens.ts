export default {
  Main: { name: 'Main' },
  Home: { name: 'Home', label: 'HOME' },
  Categories: { name: 'Categories', label: 'CATEGORIES' },
  Recipe: { name: 'Recipe' },
  RecipesList: { name: 'RecipesList' },
  Ingredient: { name: 'Ingredient' },
  IngredientsDetails: { name: 'IngredientsDetails' },
  Search: { name: 'Search', label: 'SEARCH' }
};
