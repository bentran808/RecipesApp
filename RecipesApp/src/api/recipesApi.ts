import { axiosInstance } from './axiosConfig';

export const recipesApi = {
  fetchCategoriesRequest() {
    return axiosInstance.get('categories?_embed=recipes');
  },
  fetchIngredientsRequest() {
    return axiosInstance.get('ingredients');
  },
  fetchRecipesRequest() {
    return axiosInstance.get('recipes?_expand=category');
  },
  searchByCategoryNameRequest(keyword: string) {
    return axiosInstance.get(`categories?name_like=${keyword}&_embed=recipes`);
  },
  searchByRecipeNameRequest(keyword: string) {
    return axiosInstance.get(`recipes?title_like=${keyword}`);
  }
};
