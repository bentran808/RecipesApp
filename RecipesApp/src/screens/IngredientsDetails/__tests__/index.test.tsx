import { render } from '@testing-library/react-native';
import { recipesApi } from 'api';
import Screens from 'constants/Screens';
import { ingredient, ingredientsDetails } from 'mocks';
import React from 'react';
import { Alert, TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';
import IngredientsDetailsScreen from 'screens/IngredientsDetails';

describe('Ingredients Details Screen', () => {
  let navigation: any;
  const params = { title: 'Test', ingredients: [[0, '200ml']] };

  beforeEach(() => {
    navigation = {
      navigate: jest.fn(),
      setOptions: jest.fn()
    };
    const setIngredientsDetails = jest.fn();
    jest.spyOn(React, 'useEffect').mockImplementation((f) => f());
    React.useState = jest.fn().mockReturnValue([[ingredientsDetails], setIngredientsDetails]);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should render correctly', () => {
    const tree = renderer
      .create(<IngredientsDetailsScreen navigation={navigation} route={{ params }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should call function recipesApi.fetchIngredientsRequest successful', () => {
    recipesApi.fetchIngredientsRequest = jest.fn().mockReturnValue([ingredientsDetails]);

    render(<IngredientsDetailsScreen navigation={navigation} route={{ params }} />);

    expect(recipesApi.fetchIngredientsRequest).toHaveBeenCalled();
  });

  test('should call function recipesApi.fetchIngredientsRequest failed', () => {
    jest.spyOn(Alert, 'alert');
    recipesApi.fetchIngredientsRequest = jest.fn().mockImplementation(() => {
      throw new Error('Network Error');
    });

    render(<IngredientsDetailsScreen navigation={navigation} route={{ params }} />);

    expect(Alert.alert).toHaveBeenCalledWith('Fetch data failed');
  });

  test('should call function handlePressIngredient', () => {
    const component = renderer.create(
      <IngredientsDetailsScreen navigation={navigation} route={{ params }} />
    );
    const button = component.root.findAllByType(TouchableOpacity)[0];

    button.props.onPress();

    expect(navigation.navigate).toHaveBeenCalledWith(Screens.Ingredient.name, { ingredient });
  });
});
