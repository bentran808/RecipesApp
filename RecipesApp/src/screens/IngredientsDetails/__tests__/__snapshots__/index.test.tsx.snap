// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredients Details Screen should render correctly 1`] = `
<View>
  <RCTScrollView
    data={
      Array [
        Array [
          Object {
            "id": 0,
            "name": "Oil",
            "photo_url": "https://ak7.picdn.net/shutterstock/videos/27252067/thumb/11.jpg",
          },
          "200ml",
        ],
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "height": 160,
                  "marginBottom": 10,
                  "marginLeft": 10,
                  "marginTop": 30,
                  "width": 233.33333333333334,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://ak7.picdn.net/shutterstock/videos/27252067/thumb/11.jpg",
                  }
                }
                style={
                  Object {
                    "borderRadius": 60,
                    "height": 100,
                    "width": 233.33333333333334,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontSize": 13,
                    "margin": 10,
                    "marginBottom": 5,
                    "textAlign": "center",
                  }
                }
              >
                Oil
              </Text>
              <Text
                style={
                  Object {
                    "color": "grey",
                  }
                }
              >
                200ml
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
